pkg/virt-controller/watch/vmi.go

```diff
@@ -2302,13 +2304,22 @@ func (c *VMIController) requireVolumesUpdate(vmi *virtv1.VirtualMachineInstance)
 
 func (c *VMIController) syncVolumesUpdate(vmi *virtv1.VirtualMachineInstance) {
 	vmiConditions := controller.NewVirtualMachineInstanceConditionManager()
-	condition := virtv1.VirtualMachineInstanceCondition{
-		Type:               virtv1.VirtualMachineInstanceVolumesChange,
-		LastTransitionTime: v1.Now(),
-		Status:             k8sv1.ConditionTrue,
-		Message:            "migrate volumes",
+	if vmiConditions.HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceVolumesChange, k8sv1.ConditionTrue) &&
+		migrations.MigrationFailed(vmi) {
+		vmiConditions.UpdateCondition(vmi, &virtv1.VirtualMachineInstanceCondition{
+			Type:               virtv1.VirtualMachineInstanceVolumesChange,
+			LastTransitionTime: metav1.Now(),
+			Status:             k8sv1.ConditionFalse,
+			Message:            "The migration has failed",
+		})
+	} else {
+		vmiConditions.UpdateCondition(vmi, &virtv1.VirtualMachineInstanceCondition{
+			Type:               virtv1.VirtualMachineInstanceVolumesChange,
+			LastTransitionTime: v1.Now(),
+			Status:             k8sv1.ConditionTrue,
+			Message:            "migrate volumes",
+		})
```

Comment:
Say a user fixes whatever got the migration to fail and wants it to run again.  What do they do?

Reply:
I understand that current behavior may not be great but I think an exponential backoff strategy would be better than one and done

Reply:
I'm not sure, I think if the migration fails you cannot repair it

Reply:
Why because of data on the target pvc? When a user retries migration do they also have to make sure to clean the target pvc?

Reply:
No, because the size of the destination could be wrong. In the functional test, I have created the image manually, but we can imagine that this might happen if the destination storage class has a larger overhead. Without this fix, the workload updater continuously to create migration objects and new pods using the destination pvc. I find it hard to be able to fix the storage while the update is in progress.

Additionally, it really pollutes the systems by creating pods in an infinite loop.

Reply:
> No, because the size of the destination could be wrong. 

Sure there are cases where the user configured things wrong, but what about transient errors where retrying would fix things?

> Additionally, it really pollutes the systems by creating pods in an infinite loop.

Of course which is why I am advocating for exponential retry.  That is the kubernetes way.  Transient errors will eventually converge.  If the user creates a bad config then retries will eventually happen infrequently enough to no be a big burden

Reply:
Ah, I see. I need to figure out how/where to save the number of retries

Reply:
Can you simply take advantage of built in exponential backoff when error is returned in controller reconcile loop?

Reply:
Not sure how.
The code here is actually wrong and it isn't the right place the vmi controller. I'm developing a new version moving this in the workload updater controller.